File: C2/PDOO/Practicas/DeepSpaceRuby/lib/BetaPowerEfficientSpaceStation.rb

#@author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio

#encoding:utf-8

require_relative "Dice"

module Deepspace
  class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
    
    @@EXTRAEFFICIENCY=1.2
   
    def initialize(station)
      super
      @dice=Dice.new
    end
    
    def fire
      if @dice.extraEfficiency
        factor=super*@@EXTRAEFFICIENCY
        return factor
      else return super
      end
    end
    
    def to_s
      return "BETA "+super
    end
    
  end
  
end

File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/BetaPowerEfficientSpaceStation.java


package deepspace;

/**
 *
 * @author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    
    private Dice dice;
    private static float EXTRAEFFICIENCY=1.2f;
    
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        dice=new Dice();
    }
    
    @Override
    public float fire(){
        if(dice.extraEfficiency())
            return super.fire()*EXTRAEFFICIENCY;
        else
            return super.fire();
    }
    
    @Override
    public String toString(){
        return "BETA "+super.toString();
    }
    
}
